### PR TITLE
feat(access): support sdk, merge blobstore access into cubefs sdk

### PR DESCRIPTION
**Pull request by mawei029**
**What this PR does / why we need it**:
support sdk, merge blobstore access into cubefs sdk

**Which issue this PR fixes**:

fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: blobstore/access/sdk/sdk_client.go [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+	err = transfer()
+	if err != nil {
+		access.ReportDownloadWrap(args.Location.ClusterID, "StatusOKError", "-")
+		span.Error("stream get transfer failed", errors.Detail(err))
+		return
+	}
+	span.Info("done sdk get request")
+
+	return ioutil.NopCloser(bytes.NewReader(buf)), nil
```
Comment (sejust) [inline blobstore/access/sdk/sdk_client.go:164]: io.NopCloser

File: blobstore/access/sdk/sdk_client.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+		return
+	}
+
+	if args.ReadSize > 0 && args.ReadSize != args.Location.Size {
+		span.Errorf("stream get size is invalid, bytes %d-%d/%d", args.Offset, args.Offset+args.ReadSize-1, args.Location.Size)
+		err = errcode.ErrIllegalArguments
+		return
+	}
+
+	buf := make([]byte, args.ReadSize)
```
Comment (sejust) [inline blobstore/access/sdk/sdk_client.go:147]: using stream

**Review by sejust — CHANGES_REQUESTED** [review]

File: blobstore/access/sdk/sdk_client.go [hunk 3]
```diff
@@ -240,22 +241,21 @@ func (s *sdkHandler) doDelete(ctx context.Context, args *acapi.DeleteArgs) (resp
...
-			}
-			wg.Done()
-		}(id)
-	}
+	cid := args.Locations[0].ClusterID
+	delCh := make(chan struct{}, 1)
+	go func() {
+		if err := s.handler.Delete(ctx, &acapi.Location{
+			ClusterID: cid,
+			BlobSize:  1,
```
Comment (JasonHu520) [inline blobstore/access/sdk/sdk_client.go:249]: 这个是不是可以干掉，0和1没区别似乎

File: blobstore/access/sdk/sdk_client.go [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+	for _, loc := range args.Locations { // only 1 cluster
+		if cid != loc.ClusterID {
+			return args.Locations, errcode.ErrIllegalArguments
+		}
+	}
+
+	ctx = acapi.WithReqidContextWrap(ctx)
+	locations := make([]acapi.Location, 0, len(args.Locations)) // check location size
+	for _, loc := range args.Locations {
+		if loc.Size > 0 {
```
Comment (JasonHu520) [inline blobstore/access/sdk/sdk_client.go:87]: 如果前面初始化为1，这里就不用判断咯

File: blobstore/access/stream_mock_test.go [hunk 5]
```diff
@@ -16,7 +16,7 @@ package access
 
 // github.com/cubefs/cubefs/blobstore/access/... module access interfaces
 //go:generate mockgen -destination=./controller_mock_test.go -package=access -mock_names ClusterController=MockClusterController,ServiceController=MockServiceController,VolumeGetter=MockVolumeGetter github.com/cubefs/cubefs/blobstore/access/controller ClusterController,ServiceController,VolumeGetter
-//go:generate mockgen -destination=./access_mock_test.go -package=access -mock_names StreamHandler=MockStreamHandler,Limiter=MockLimiter github.com/cubefs/cubefs/blobstore/access StreamHandler,Limiter
+//go:generate mockgen -destination=../testing/mocks/access_stream.go -package=mocks -mock_names StreamHandler=MockStreamHandler github.com/cubefs/cubefs/blobstore/access StreamHandler
```
Comment (JasonHu520) [inline blobstore/access/stream_mock_test.go:19]: 目标文件可以加个mock前缀

Comment (sejust) [reply blobstore/access/stream_mock_test.go:19]: 放到testing目录下的不用加，重复了

**Pull request closed without merging**